zcash_primitives/Cargo.toml

```diff
@@ -15,6 +15,7 @@ fpe = "0.1"
 hex = "0.3"
 lazy_static = "1"
 pairing = { path = "../pairing" }
+rand = "0.7"
```

Comment:
I added a dependency on `rand` separately from `rand_core` and `rand_os` because I wanted to access `rand::seq::SliceRandom` for shuffling the order of spends and outputs. I didn't bother replacing all the usages of `rand_core` and `rand_os` because it is unnecessary noise.